pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1250,6 +1250,139 @@ def _controlled(self, wire):
+
+    @staticmethod
+    def compute_matrix():  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.ECR.matrix`
+
+
+        Return type: tensor_like
+
+        **Example**
+
+        >>> print(qml.ECR.compute_matrix())
+
+         [[0+0.j 0.+0.j 1/sqrt(2)+0.j 0.+1j/sqrt(2)]
+         [0.+0.j 0.+0.j 0.+1.j/sqrt(2) 1/sqrt(2)+0.j]
+         [1/sqrt(2)+0.j 0.-1.j/sqrt(2) 0.+0.j 0.+0.j]
+         [0.-1/sqrt(2)j 1/sqrt(2)+0.j 0.+0.j 0.+0.j]]"""
+        return np.array(
+            [
+                [0, 0, INV_SQRT2, INV_SQRT2 * 1j],
+                [0, 0, INV_SQRT2 * 1j, INV_SQRT2],
+                [INV_SQRT2, -INV_SQRT2 * 1j, 0, 0],
+                [-INV_SQRT2 * 1j, INV_SQRT2, 0, 0],
+            ]
+        )
+
+    @staticmethod
+    def compute_eigvals():
+        r"""Eigenvalues of the operator in the computational basis (static method).
+
+        If :attr:`diagonalizing_gates` are specified and implement a unitary :math:`U`,
+        the operator can be reconstructed as
+
+        .. math:: O = U \Sigma U^{\dagger},
+
+        where :math:`\Sigma` is the diagonal matrix containing the eigenvalues.
+
+        Otherwise, no particular order for the eigenvalues is guaranteed.
+
+        .. seealso:: :meth:`~.ECR.eigvals`
+
+
+        Returns:
+            array: eigenvalues
+
+        **Example**
+
+        >>> print(qml.ECR.compute_eigvals())
+        [1, -1, 1, -1]
+        """
+
+        return np.array([1, -1, 1, -1])
+
+    @staticmethod
+    def compute_decomposition(wires):
+        r"""Representation of the operator as a product of other operators (static method).
+
+           .. math:: O = O_1 O_2 \dots O_n.
+
+
+           .. seealso:: :meth:`~.ECR.decomposition`.
+
+           Args:
+               wires (Iterable, Wires): wires that the operator acts on
+
+           Returns:
+               list[Operator]: decomposition into lower level operations
+
+           **Example:**
+
+           >>> print(qml.ECR.compute_decomposition((0,1)))
+
+
+        [PauliZ(wires=[0]),
+         CNOT(wires=[0, 1]),
+         SX(wires=[1]),
+         RX(1.5707963267948966, wires=[0]),
+         RY(1.5707963267948966, wires=[0]),
+         RX(1.5707963267948966, wires=[0])]
+
+        """
+        pi = np.pi
+        return [
+            PauliZ(wires=[wires[0]]),
+            CNOT(wires=[wires[0], wires[1]]),
+            SX(wires=[wires[1]]),
+            qml.RX(pi / 2, wires=[wires[0]]),
+            qml.RY(pi / 2, wires=[wires[0]]),
+            qml.RX(pi / 2, wires=[wires[0]]),
+        ]
+
+    def adjoint(self):
+        op = ECR(wires=self.wires)
+        op.inverse = not self.inverse
+        return op
```

Comment:
Hi @avbhadw ,
It looks like this gate is it's own adjoint, in which case you can just return a new, uninverted op.

```suggestion
    def adjoint(self):
        return ECR(wires=self.wires)
```

We are currently trying to move away from relying on in-place inversion as well.

You can test this method by simply adding it to the [`involution_ops`](https://github.com/PennyLaneAI/pennylane/blob/89439a811a5d5858b86b509923eaefd72a14391d/tests/ops/qubit/test_non_parametric_ops.py#L1187) list near the bottom of `tests/ops/qubit/test_non_parametric_ops.py`.

Reply:
Thanks for the updates! I'll defer to @antalszava for everything else in the PR.